tool/cli.js

```diff
@@ -46,14 +46,36 @@ program
   .name("tool")
   .version("0.0.0")
   .disableGlobalOption("--silent")
-  .command("validate-redirects", "Check the _redirects.txt file(s)")
+  .command("validate-redirects", "Try loading the _redirects.txt file(s)")
   .action(
     tryOrExit(({ logger }) => {
       Redirect.load(null, true);
       logger.info(chalk.green("🍾 All is well in the world of redirects 🥂"));
     })
   )
 
+  .command("verify-redirects", "Verify the _redirects.txt file(s) by locale")
+  .argument("[locale...]", "Locale", {
+    default: [...VALID_LOCALES.values()],
+    validator: [...VALID_LOCALES.values()],
+  })
+  .action(
+    tryOrExit(({ args, logger }) => {
+      const { locale } = args;
+
+      for (const l of locale) {
+        try {
+          Redirect.validateLocale(l);
+          logger.info(chalk.green(`✓ redirects for ${l} looking good!`));
+        } catch (e) {
+          logger.info(
+            chalk.green(`_redirects.txt for ${l} is causing issues: ${e}`)
+          );
+        }
+      }
+    })
+  )
```

Comment:
It's so confusing. `validate-redirects` doesn't validate the redirects??
"Verify" is when you check something after you've done something.

Reply:
https://en.wikipedia.org/wiki/Software_verification_and_validation

I don't know how to name these. I guess right now:
*validate-redirects* asserts that the (all) `_redirects.txt` can be loaded and does it's own basic sanity check.

*verify-redirects* asserts that a (many) `_redirects.txt` have optimized and correct entries by calling `validateTo(From)URL` for each entry, checking encoding and asserts that it's in alphabetical order and optimized.

We should run *validate-redirects* during deployment and yari PRs and *verify-redirects* for content PRs.

Happy to change the name or move to change the `load` function to also _verify_.

Reply:
From the Wikipedia article:

- `verify` - check that it worked
- `validate` - was it the right check?

I think some guidance is required. The fact that we had to discuss the meaning of the words indicates to me that this is potentially too complicated. (Also, when you skim the code it says that the function is called "verify-redirects" and inside it calls "validateLocale()")

Why does any CLI need to be run "during deployment"?

What we want is two tools:

- One that we can run in content pull requests CI to make sure that the contents of the `_redirects.txt` file isn't corrupt or imperfect. I.e. guarding against people manually messing with the .txt file without. 
- One that we only use very rarely, by the core dev team, to tidy and clean up the `_redirects.txt` file so it's above-and-beyond sane and optimized as some redirects can be short-circuited to perfection. 

Perhaps take one last look at the names and the help strings to see if they can be optimized (for usability/understanding) so they can be easily understood by us in 6 months times (or a new dev joining tomorrow) can understand them without having to break down the code.